base/broadcast.jl

```diff
@@ -10,6 +10,11 @@ export broadcast_getindex, broadcast_setindex!
 
 ## Broadcasting utilities ##
 
+# fallback routines for broadcasting with no arguments or with scalars
+# to just produce a scalar result:
+broadcast(f) = f()
+broadcast(f, x::Number...) = f(x...)
```

Comment:
I think this is ambiguous with https://github.com/JuliaLang/julia/blob/2b98dda496760466a07225ae5ef0341aafbbaec8/base/deprecated.jl#L370 (but not detected? cc @timholy) and causes, for example `speye(5).(2)` to not trigger the deprecation `getfield` path that it would have meant on 0.4, but instead trigger a `MethodError: objects of type SparseMatrixCSC{Float64,Int32} are not callable`.